relay-server/src/services/spooler/mod.rs

```diff
@@ -526,7 +527,7 @@ impl OnDisk {
         loop {
             // Before querying the db, make sure that the buffer guard has enough availability:
             self.dequeue_attempts += 1;
-            if !self.buffer_guard.is_below_low_watermark() {
+            if !self.memory_stat_config.has_enough_memory() {
```

Comment:
In this new version of the spooler, we are just checking a single memory level to validate whether we want to perform certain actions. This will change once the new spooler will be implemented.